docs/i18n-l10n/translating-text-strings.md

```diff
@@ -0,0 +1,447 @@
+
+# Use inherited translate() function to get the final text string
+translated = translate(msgid, context=getRequest())
+
+# translated is now u"Käännetty teksti" (in Finnish)
+```
+
+
+(translating-text-strings-non-python-message-ids-label)=
+
+### Non-Python message IDs
+
+There are other message ID markers in code outside the Python domain that have their own mechanisms:
+
+-   ZCML entries
+-   GenericSetup XML
+-   TAL page templates
+
+
+(translating-text-strings-translation-string-substitution-label)=
+
+## Translation string substitution
+
+Translation string substitutions must be used when the final translated message contains variable strings.
+
+The translation domain will be taken from the `msgid` object itself, which is a string-like `zope.i18nmessageid` instance.
+
+Message IDs are immutable (read-only) objects.
+You need to always create a new message ID if you use different variable substitution mappings.
+For example, in your view code, you would do the following.
+
+```python
+from saariselka.app import _
+from zope.i18n import translate
+from zope.globalrequest import getRequest
+
+class SomeView(BrowserView):
+
+    def do_stuff(self):
+
+        msgid = _("search_results_found_msg", default=u"Found ${results} results", mapping={u"results": len(self.contents)})
+
+        # Use inherited translate() function to get the final text string
+        translated = translate(msgid, context=getRequest())
+
+        # Show the final result count to the user as a portal status message
+        messages = IStatusMessage(self.request)
+        messages.addStatusMessage(translated, type="info")
+```
+
+
+(translating-text-strings-i18ndude-label)=
+
+## `i18ndude`
+
+{term}`i18ndude` is a developer-oriented command-line utility to manage PO and MO files.
+
+Usually you build our own shell script wrapper around `i18ndude` to automate generation of MO files from your package's PO files.
+
+Plone will automatically compile all PO files to MO files on start up if a specific environment variable is enabled.
+
+In your `buildout.cfg` in the part using `plone.recipe.zope2instance`, you can set an environment variable for this.
+
+```ini
+environment-vars =
+    zope_i18n_compile_mo_files true
+```
+
+Note that the value does not matter.
+The code in `zope.i18n` looks for the mere existence of the variable, and does not care what is its value.
+
+If you do not add that environment variable, you will need to provide the MO files in your package.
+To make this easier, and if you use [zest.releaser](https://pypi.org/project/zest.releaser/) to publish your packages, you can use [zest.pocompile](https://pypi.org/project/zest.pocompile/).
+This script hooks into the release process and builds the MO files for you.
+
+
+(translating-text-strings-installing-i18ndude-label)=
+
+### Installing i18ndude
+
+The recommended method is to have {term}`i18ndude` installed via your [buildout](https://www.buildout.org/en/latest/).
```

Comment:
I am still a fan of buildout, but here I would say: use `bin/pip install i18ndude`. I don't mind though.

Reply:
Is the future direction away from buildout to pip?

These docs refer to buildout in several places, mostly because that was what was used when they were first written many years ago. Should they also be changed to use pip?

Sorry, I am a novice with buildout, and have used it only a few times.

Reply:
It's fine to keep the buildout reference then. Those parts can be rewritten later if needed.